Some tests fail on arm64

### Bug Description

Some tests fail on arm64 due to having the architecture hard-coded into expected values.

### To Reproduce

Run tests on arm64

### part yaml

_No response_

### Relevant log output

```shell
=================================== FAILURES ===================================
_________________________________ test_install _________________________________

fake_etc_apt = PosixPath('/tmp/pytest-of-ubuntu/pytest-0/test_install0/etc/apt')
all_repo_types = [{'components': ['main'], 'formats': ['deb', 'deb-src'], 'key-id': '78E1918602959B9C59103100F1831DDAFC42E99D', 'priori..., 'suites': ['focal'], 'type': 'apt', ...}, {'cloud': 'antelope', 'pocket': 'updates', 'priority': 123, 'type': 'apt'}]
test_keys_dir = PosixPath('/tmp/pytest-of-ubuntu/pytest-0/test_install0/keys')

    @pytest.mark.slow
    def test_install(fake_etc_apt, all_repo_types, test_keys_dir):
        """Integrated test that checks the configuration of keys, sources and pins."""
    
        assert repo.install(project_repositories=all_repo_types, key_assets=test_keys_dir)
    
        check_keyrings(fake_etc_apt)
>       check_sources(fake_etc_apt, signed_by_location=fake_etc_apt / "keyrings")

tests/integration/repo/test_installer.py:205: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

etc_apt_dir = PosixPath('/tmp/pytest-of-ubuntu/pytest-0/test_install0/etc/apt')
signed_by_location = PosixPath('/tmp/pytest-of-ubuntu/pytest-0/test_install0/etc/apt/keyrings')

    def check_sources(etc_apt_dir: Path, signed_by_location: Path) -> None:
        sources_dir = etc_apt_dir / "sources.list.d"
    
        keyrings_on_fs = etc_apt_dir / "keyrings"
    
        if CLOUD_DATA:
            cloud_name = CLOUD_DATA["cloud"]
            codename = CLOUD_DATA["codename"]
        else:
            cloud_name = codename = None
    
        # Must have exactly these sources files, one for each repo
        source_to_contents = {
            "http_ppa_launchpad_net_snappy_dev_snapcraft_daily_ubuntu": APT_SOURCES.format(
                key_location=signed_by_location
            ),
            "ppa-deadsnakes_ppa": PPA_SOURCES.format(
                codename=VERSION_CODENAME, key_location=signed_by_location
            ),
            "http_apt_puppet_com": PUPPET_SOURCES.format(key_location=signed_by_location),
        }
        if CLOUD_DATA:
            source_to_contents[f"cloud-{cloud_name}"] = CLOUD_SOURCES.format(
                cloud=cloud_name,
                codename=codename,
                key_location=signed_by_location,
            )
    
        assert len(list(keyrings_on_fs.iterdir())) == len(source_to_contents)
    
        for source_repo, expected_contents in source_to_contents.items():
            source_file = sources_dir / f"craft-{source_repo}.sources"
            assert source_file.is_file()
>           assert source_file.read_text() == expected_contents
E           AssertionError: assert 'Types: deb d...C42E99D.gpg\n' == 'Types: deb d...C42E99D.gpg\n'
E               Types: deb deb-src
E               URIs: http://ppa.launchpad.net/snappy-dev/snapcraft-daily/ubuntu
E               Suites: focal
E               Components: main
E             - Architectures: amd64
E             ?                  -
E             + Architectures: arm64
E             ?                 +
E               Signed-By: /tmp/pytest-of-ubuntu/pytest-0/test_install0/etc/apt/keyrings/craft-FC42E99D.gpg

tests/integration/repo/test_installer.py:275: AssertionError
_____________________________ test_install_in_root _____________________________

tmp_path = PosixPath('/tmp/pytest-of-ubuntu/pytest-0/test_install_in_root0')
all_repo_types = [{'components': ['main'], 'formats': ['deb', 'deb-src'], 'key-id': '78E1918602959B9C59103100F1831DDAFC42E99D', 'priori..., 'suites': ['focal'], 'type': 'apt', ...}, {'cloud': 'antelope', 'pocket': 'updates', 'priority': 123, 'type': 'apt'}]
test_keys_dir = PosixPath('/tmp/pytest-of-ubuntu/pytest-0/test_install_in_root0/keys')

    @pytest.mark.slow
    def test_install_in_root(tmp_path, all_repo_types, test_keys_dir):
        """Integrated test that checks the configuration of keys, sources and pins."""
        etc_apt = tmp_path / "etc/apt"
        create_etc_apt_dirs(etc_apt)
    
        assert repo.install_in_root(
            project_repositories=all_repo_types, key_assets=test_keys_dir, root=tmp_path
        )
    
        check_keyrings(etc_apt)
>       check_sources(etc_apt, signed_by_location=Path("/etc/apt/keyrings"))

tests/integration/repo/test_installer.py:220: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

etc_apt_dir = PosixPath('/tmp/pytest-of-ubuntu/pytest-0/test_install_in_root0/etc/apt')
signed_by_location = PosixPath('/etc/apt/keyrings')

    def check_sources(etc_apt_dir: Path, signed_by_location: Path) -> None:
        sources_dir = etc_apt_dir / "sources.list.d"
    
        keyrings_on_fs = etc_apt_dir / "keyrings"
    
        if CLOUD_DATA:
            cloud_name = CLOUD_DATA["cloud"]
            codename = CLOUD_DATA["codename"]
        else:
            cloud_name = codename = None
    
        # Must have exactly these sources files, one for each repo
        source_to_contents = {
            "http_ppa_launchpad_net_snappy_dev_snapcraft_daily_ubuntu": APT_SOURCES.format(
                key_location=signed_by_location
            ),
            "ppa-deadsnakes_ppa": PPA_SOURCES.format(
                codename=VERSION_CODENAME, key_location=signed_by_location
            ),
            "http_apt_puppet_com": PUPPET_SOURCES.format(key_location=signed_by_location),
        }
        if CLOUD_DATA:
            source_to_contents[f"cloud-{cloud_name}"] = CLOUD_SOURCES.format(
                cloud=cloud_name,
                codename=codename,
                key_location=signed_by_location,
            )
    
        assert len(list(keyrings_on_fs.iterdir())) == len(source_to_contents)
    
        for source_repo, expected_contents in source_to_contents.items():
            source_file = sources_dir / f"craft-{source_repo}.sources"
            assert source_file.is_file()
>           assert source_file.read_text() == expected_contents
E           AssertionError: assert 'Types: deb d...C42E99D.gpg\n' == 'Types: deb d...C42E99D.gpg\n'
E               Types: deb deb-src
E               URIs: http://ppa.launchpad.net/snappy-dev/snapcraft-daily/ubuntu
E               Suites: focal
E               Components: main
E             - Architectures: amd64
E             ?                  -
E             + Architectures: arm64
E             ?                 +
E               Signed-By: /etc/apt/keyrings/craft-FC42E99D.gpg

tests/integration/repo/test_installer.py:275: AssertionError
================= 2 failed, 1 passed, 218 deselected in 58.70s =================
```

--- a/.github/workflows/qa.yaml
+++ b/.github/workflows/qa.yaml
@@ -15,8 +15,7 @@ jobs:
   test:
     uses: canonical/starflow/.github/workflows/test-python.yaml@main
     with:
-      # Limiting to amd64 as a workaround for https://github.com/canonical/craft-archives/issues/144
-      # When fixing that bug, add arm64 runners as well for fast tests.
-      fast-test-platforms: '[["jammy", "amd64"], ["noble", "amd64"]]'
-      slow-test-platforms: '[["jammy", "amd64"]]'
-      lowest-python-platform: self-hosted-linux-amd64-jammy-edge
+      # Testing on both amd64 and arm64 to prevent architecture-specific bugs.
+      fast-test-platforms: '["jammy", ["noble", "amd64"], ["noble", "arm64"]]'
+      slow-test-platforms: '[["jammy", "amd64"], ["jammy", "arm64"]]'
+      lowest-python-platform: jammy

--- a/tests/integration/repo/test_installer.py
+++ b/tests/integration/repo/test_installer.py
@@ -23,7 +23,7 @@ from typing import Any, Dict, List
 
 import distro
 import pytest
-from craft_archives import repo
+from craft_archives import repo, utils
 
 APT_SOURCES = dedent(
     """
@@ -31,7 +31,7 @@ APT_SOURCES = dedent(
     URIs: http://ppa.launchpad.net/snappy-dev/snapcraft-daily/ubuntu
     Suites: focal
     Components: main
-    Architectures: amd64
+    Architectures: {host_arch}
     Signed-By: {key_location}/craft-FC42E99D.gpg
     """
 ).lstrip()
@@ -44,7 +44,7 @@ PPA_SOURCES = dedent(
     URIs: http://ppa.launchpad.net/deadsnakes/ppa/ubuntu
     Suites: {codename}
     Components: main
-    Architectures: amd64
+    Architectures: {host_arch}
     Signed-By: {key_location}/craft-6A755776.gpg
     """
 ).lstrip()
@@ -65,7 +65,7 @@ CLOUD_SOURCES = dedent(
     URIs: http://ubuntu-cloud.archive.canonical.com/ubuntu
     Suites: {codename}-updates/{cloud}
     Components: main
-    Architectures: amd64
+    Architectures: {host_arch}
     Signed-By: {key_location}/craft-EC4926EA.gpg
     """
 ).lstrip()
@@ -76,7 +76,7 @@ PUPPET_SOURCES = dedent(
     URIs: http://apt.puppet.com
     Suites: focal
     Components: puppet-tools
-    Architectures: amd64
+    Architectures: {host_arch}
     Signed-By: {key_location}/craft-9E61EF26.gpg
     """
 ).lstrip()
@@ -253,18 +253,25 @@ def check_sources(etc_apt_dir: Path, signed_by_location: Path) -> None:
     # Must have exactly these sources files, one for each repo
     source_to_contents = {
         "http_ppa_launchpad_net_snappy_dev_snapcraft_daily_ubuntu": APT_SOURCES.format(
-            key_location=signed_by_location
+            key_location=signed_by_location,
+            host_arch=utils.get_host_architecture(),
         ),
         "ppa-deadsnakes_ppa": PPA_SOURCES.format(
-            codename=VERSION_CODENAME, key_location=signed_by_location
+            codename=VERSION_CODENAME,
+            key_location=signed_by_location,
+            host_arch=utils.get_host_architecture(),
         ),
-        "http_apt_puppet_com": PUPPET_SOURCES.format(key_location=signed_by_location),
+        "http_apt_puppet_com": PUPPET_SOURCES.format(
+            key_location=signed_by_location,
+            host_arch=utils.get_host_architecture(),
+        ),
     }
     if CLOUD_DATA:
         source_to_contents[f"cloud-{cloud_name}"] = CLOUD_SOURCES.format(
             cloud=cloud_name,
             codename=codename,
             key_location=signed_by_location,
+            host_arch=utils.get_host_architecture(),
         )
 
     assert len(list(keyrings_on_fs.iterdir())) == len(source_to_contents)

